server/configuration.go

```diff
@@ -1,31 +1,70 @@
 package main
 
-import "errors"
+import (
+	"github.com/pkg/errors"
+)
 
-type Config struct {
+// configuration captures the plugin's external configuration as exposed in the Mattermost server
+// configuration, as well as values computed from the configuration. Any public fields will be
+// deserialized from the Mattermost server configuration in OnConfigurationChange.
+type configuration struct {
 	Trigger string
 }
 
+// OnConfigurationChange loads the plugin configuration, validates it and saves it.
 func (p *MatterpollPlugin) OnConfigurationChange() error {
-	c := &Config{}
-	if err := p.API.LoadPluginConfiguration(c); err != nil {
-		return err
+	configuration := new(configuration)
+
+	if err := p.API.LoadPluginConfiguration(configuration); err != nil {
+		return errors.Wrap(err, "failed to load plugin configuration")
 	}
 
-	if c.Trigger == "" {
+	if configuration.Trigger == "" {
 		return errors.New("Empty trigger not allowed")
 	}
 
-	if p.Config != nil {
-		if err := p.API.UnregisterCommand("", p.Config.Trigger); err != nil {
-			return err
+	if p.getConfiguration().Trigger != "" {
```

Comment:
Should we avoid to call `getConfiguration` twice here?
Or `OnConfigurationChange` isn't called concurrently?